src/android/CameraPreviewFragment.java

```diff
@@ -222,27 +231,72 @@ public void deviceHasUltraWideCamera(HasUltraWideCameraCallback hasUltraWideCame
         hasUltraWideCameraCallback.onResult(defaultCamera == true && ultraWideCamera == true);
     }
 
-    public static Size calculateResolution(Context context, int targetSize) {
-        Size calculatedSize;
-        if (getScreenOrientation(context) == Configuration.ORIENTATION_PORTRAIT) {
-            calculatedSize = new Size((int) ((float) targetSize / ratio), targetSize);
-        } else {
-            calculatedSize = new Size(targetSize, (int) ((float) targetSize / ratio));
+     public static Size calculateResolution(Context context, int desiredWidthPx, double aspectRatio) {
+        int minWidthPx = 700;
+        // 1) get all supported JPEG output sizes
```

Comment:
```suggestion
        // 1) get all supported JPEG output sizes
```
Cleanup the comments a bit